workshop/content/basics/operations/troubleshooting-provisioning-errors/_index.md

```diff
@@ -0,0 +1,116 @@
+---
+title: 'Troubleshooting provisioning errors'
+date: 2022-01-04T19:00:24Z
+weight: 500
+---
+
+### Overview
+As you iterate on the development of your CloudFormation template, you test provisioning of resources you describe in your template by creating a CloudFormation [stack](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacks.html). If you specify incorrect property values for resource configurations in your template, by default your stack will roll back to the last-known stable state, and all stack resources will be rolled back. Let's take an example where you create a stack off of a template, in which you describe 10 resources. In this example, 9 resources you described are successfully created, and the creation of the tenth resource fails: by default, your stack will roll back, including the 9 resources that were successfully provisioned.
+
+To speed up development cycles, you can choose to [preserve](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stack-failure-options.html) the state of resources that have been successfully provisioned as part of stack create and update operations, and of [change set](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-changesets.html) operations. When you choose to use this functionality, you pause the stack rollback to preserve the state of successfully provisioned resources, and you can focus on troubleshooting and fixing your configurations, so that you can resume your provisioning operations when you are ready.
+
+### Topics Covered
+By the end of this lab, you will be able to:
+
+* Understand how to troubleshoot provisioning errors, whilst preserving the state resources you successfully deployed
+* Navigate the [AWS resource and property types reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html) to discover resource properties, and return values for a given resource, from which you can choose as you develop your template
+
+### Start Lab
+You will use an example CloudFormation template, that contains an incorrect resource configuration, to create a new stack. You will choose to *preserve successfully provisioned resources* as part of *stack failure options* to preserve the state of an example `ExampleDeadLetterQueue` [Amazon SQS](https://aws.amazon.com/sqs/) queue resource, that will be successfully provisioned as part of the stack creation operation.
+
+The creation operation for your stack will fail because another SQS queue described in your template, `ExampleSourceQueue`, has a configuration error. You will troubleshoot and fix the error in the template, and then you will choose to resume the stack creation operation with the template you updated.
```

Comment:
```suggestion
The creation operation for the stack will fail because another SQS queue is described in the template, and has a configuration error. Once you troubleshoot and fix the error in the template, you can resume the stack creation operation with the updated template.
```